test/extended/olm/olmv1-webhook-support.go

```diff
@@ -0,0 +1,261 @@
+package operators
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"path/filepath"
+	"time"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	conditionTypeServing   = "Serving"
+	conditionTypeInstalled = "Installed"
+)
+
+var (
+	webhookTestBaseDir = exutil.FixturePath("testdata", "olmv1", "webhook-support")
+
+	// should agree with the values in the resources in testdata/olmv1/webhook-support
+	testNamespace                     = "webhook-operator"
+	mutatingWebhookTestResourceName   = "mutating-webhook-test"
+	conversionWebhookTestResourceName = "conversion-webhook-test"
+	webhookServiceName                = "webhook-operator-webhook-service"
+	certificateSecretName             = "webhook-operator-webhook-service-cert"
+)
+
+var _ = g.Describe("[sig-olmv1][OCPFeatureGate:NewOLMWebhookProviderOpenshiftServiceCA][Skipped:Disconnected][Serial] OLMv1 operator with webhooks", g.Ordered, g.Serial, func() {
```

Comment:
I wonder if there is some way to test certificate rotation and check that the service is available _after_ a rotation occurs? I feel like that has been problematic in OLMv0's implementation.

Reply:
I've updated the two tests: 
- ensuring service availability during cert secret deletion by continuously creating and deleting a resource after the deletion of the cert secret
- ensuring deployment availability during cert rotation, which is forced to happen when the openshift-service-ca signing key gets deleted. The problem I ran into is that then the certificate is no longer signed and fails for another reason. So, I'm not sure the best way to test this case =S maybe we just cannot...

Reply:
For the latter case, I think my expectation would be:
1. If service-ca signing key is deleted, then service-ca-operator creates a new signing key and then rotates all certs.
2. When service-ca-operator creates a new signing key, operator-controller notices and updates its CA store used to validate the catalogd server cert.
3. When service-ca-operator rotates catalogd's cert (so that it is signed by the new signing key), the catalogd pods notice that change and start using that new cert.

2 and 3 wouldn't happen atomically at the same time, so I think it would be safe to assume _some_ downtime, between catalogd and operator-controller, but then for that to correct itself on its own after _both_ (2) and (3) occur.

Is that your understanding of what should happen?

Reply:
I think that's reasonable. I was wondering if it would be a controlled hand-off to the new pod by the deployment.
I.e. when the tls secret changes, deployment notices, rotates the pods by bringing up the new, switching traffic, then bringing down the old one. But, it doesn't sound unreasonable that there might be some downtime.

Reply:
The service-ca validity period is 26 months, a time period where users are expected to have upgraded, such that the certificate is rotated as part of an upgrade.